util/build_configs/cray-internal/setenv-xc-x86_64.bash

```diff
@@ -0,0 +1,468 @@
+            # - local workarounds such as the variables shown in the "venv" callback, below.
+
+        log_info "Start build_configs $dry_run $verbose -- $venv_targets"
+
+        $cwd/../build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.venv.log" \
+            --target-compiler=venv -- $venv_targets
+
+        if [ "$pyLT27" = "True" ]; then
+
+            # If the system-installed Python was 2.6, Build a second copy of Chapel
+            # python-venv tools using the alternate Python 2.7 installation,
+            # which must be implemented by the "venv_py27" callback, below.
+
+            log_info "Building Chapel component: venv ($venv_targets with venv_py27 callback)"
+
+            log_info "Start build_configs $dry_run $verbose -- $venv_targets"
+
+            $cwd/../build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.venv_py27.log" \
+                --target-compiler=venv_py27 -- $venv_targets
+
+            # Custom Chapel make target to rm files or links named "python" from the installed py27
+            # bin dir, forcing users of the installed Chapel RPM to get "python" from the system.
+
+            use_system_python="-C third-party/chpl-venv use-system-python"
+
+            log_info "Start build_configs $dry_run $verbose -- $use_system_python"
+
+            $cwd/../build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.venv_py27-use_system_python.log" \
+                --target-compiler=venv_py27 -- $use_system_python
+        fi
+        ;;
+    ( * )
+        log_info "NO building Chapel component: venv"
+        ;;
+    esac
+
+    case "$components" in
+    ( *mason* )
+        log_info "Building Chapel component: mason"
+
+        log_info "Start build_configs $dry_run $verbose -- mason"
+
+        $cwd/../build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.mason.log" \
+            --target-compiler=gnu -- mason
+        ;;
+    ( * )
+        log_info "NO building Chapel component: mason"
+        ;;
+    esac
+
+    clean_targets="cleanall"
+    case "$components" in
+    ( *clean* )
+        log_info "Building Chapel component: clean (make $clean_targets)"
+
+        log_info "Start build_configs $dry_run $verbose -- $clean_targets"
+
+        $cwd/../build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.clean.log" \
+            --target-compiler=gnu -- $clean_targets
+        ;;
+    ( * )
+        log_info "NO building Chapel component: clean"
+        ;;
+    esac
+
+    log_info "End $setenv"
+else
+
+# ===============
+# setenv callback
+# ===============
+
+    setenv=$( basename "${BASH_SOURCE[0]}" )
+    cwd=$( cd $(dirname "${BASH_SOURCE[0]}") && pwd )
+    source $cwd/../functions.bash
+
+    log_info "Begin callback $setenv"
+    log_debug "with config=$BUILD_CONFIGS_CALLBACK"
+
+    # Exit immediately to skip (avoid building) unwanted Chapel configs
+
+    if [ "$CHPL_COMM" == ugni ]; then
```

Comment:
Maybe take a poll: for the module, should we build any multi-node config (comm!=none, not just comm=ugni) without a launcher?  I'd be inclined to say we could skip that combination.

Reply:
As noted elsewhere, the original goal was to repro the existing Chapel Cray modules using the new build scripts... by that reasoning, it was "wrong" for these setenvs to skip the launcher=none build for comm==ugni, let alone for the other comms.  

I think it happened because the ancestor of these setenv scripts (setenv-example3 in the parent directory) took this approach, and I've never seen any harm come of it. 

Open for advice.

Reply:
It's fine with me if this stays as it is for now.  My comment really had to do with the general desirability of the combination comm!=none and launcher=none in the module.  If the point of the current work is to precisely reproduce the builds as they are today then it doesn't need to be addressed now.